openlibrary/plugins/openlibrary/js/merge-request-table/index.js

```diff
@@ -277,3 +277,96 @@ function toggleMergeLink(mergeLink) {
         mergeLink.classList.toggle('hidden')
     }
 }
+
+/**
+ * Toggle a dropdown menu while closing other dropdown menus.
+ *
+ * @param {Event} event
+ * @param {string} menuButtonId
+ */
+function toggleAMenuWhileClosingOthers(event, menuButtonId) {
+    // prevent closing of menu on bubbling unless click menuButton itself
+    if (event.target.id === menuButtonId) {
+        // close other open menus, then toggle selected menu
+        closeOtherMenus(menuButtonId)
+        event.target.firstElementChild.classList.toggle('hidden')
+    }
+}
+
+/**
+ * Close dropdown menus whose menu button doesn't match a given id.
+ *
+ * @param {string} menuButtonId
+ */
+function closeOtherMenus(menuButtonId) {
+    const dropMenus = document.querySelectorAll('.mr-dropdown')
+    dropMenus.forEach((menuButton) => {
+        if (menuButton.id !== menuButtonId) {
+            menuButton.firstElementChild.classList.add('hidden')
+        }
+    })
+}
+
+/**
+ * Filters of dropdown menu items using case-insensitive string matching.
+ *
+ * @param {Event} event
+ */
+function filterMenuItems(event) {
+    const input = document.getElementById(event.target.id)
+    const filter = input.value.toUpperCase()
+    const menu = input.parentElement
+    const items = menu.getElementsByClassName('dropdown-item')
+    // skip first item in menu
+    for (let i=1; i < items.length; i++) {
+        const text = items[i].textContent
+        if (text.toUpperCase().indexOf(filter) > -1) {
+            items[i].style.display = ''
+        }
+        else {
+            items[i].style.display = 'none'
+        }
+    }
+}
+
+/**
+ * Close all dropdown menus when click anywhere on screen that's not part of
+ * the dropdown menu; otherwise, keep dropdown menu open.
+ *
+ * @param {Event} event
+ */
+function closeMenusIfClickOutside(event) {
+    const dropButtons = Array.from(document.querySelectorAll('.mr-dropdown'))
+    const menusClicked = dropButtons.filter((menuButton) => {
+        return menuButton.contains(event.target)
+    })
+    // want to preserve clicking in a menu, i.e. when filtering for users
+    if (!menusClicked.length) {
+        const dropMenus = document.querySelectorAll('.mr-dropdown-menu')
+        dropMenus.forEach((menu) => menu.classList.add('hidden'))
+    }
+}
+
+/**
+ * Initializes events for merge queue filter dropdown menu functionality.
+ *
+ */
+export function initFilters() {
+    const dropMenus = document.querySelectorAll('.mr-dropdown')
+    dropMenus.forEach((menu) => {
+        menu.addEventListener('click', (event) => {
+            toggleAMenuWhileClosingOthers(event, menu.id)
+        })
+    })
+    const closeButtons = document.querySelectorAll('.dropdown-close')
+    closeButtons.forEach((button) => {
+        button.addEventListener('click', (event) => {
+            event.target.closest('div').classList.toggle('hidden')
```

Comment:
```suggestion
            event.target.closest('.mr-dropdown-menu').classList.toggle('hidden')
```
If the close button is, in the future, nested within another `div`, this will break.  Best to use a class or ID selector in cases like these.